### PR TITLE
Demonstration of get all document feature.

### PR DESCRIPTION
**Pull request by arnaudgelas**
TODO: tests must be fixed

### CODE REVIEW

**Comment by arnaudgelas** [comment]
@verthais Do you better understand the document thing with these tests?

File: backend/tests/test_api.py [hunk 1]
```diff
@@ -253,60 +254,62 @@ def test_get_all_sdes_owned_by_given_user():
...
+    koala_api.add_document(doc_a)
+    koala_api.add_document(doc_b)
+
     koala_api.add_tool(clang)
-    ide = Tool(name="ide", version_major=12, purpose="IDE")
-    koala_api.add_tool(ide)
+    koala_api.add_system_document(clang, doc_a)
+    koala_api.add_system_document(clang, doc_b)
+
+    result = koala_api.get_tool_documents(clang)
```
Comment (arnaudgelas) [inline backend/tests/test_api.py:291]: For the sake of consistency I would also rename `get_tool_documents` -and `get_system_documents` > `get_linked_documents`

File: backend/koala/api.py [hunk 2]
```diff
@@ -46,21 +62,50 @@ def add_system(self, system: System) -> None:
...
+    def add_system_document(self, system: System, document: Document) -> None:
+        pass
+
+    def get_system_documents(self, system: System) -> List[Document]:
+        pass
+
     @staticmethod
-    def _convert_to(cls, tool_database: List[ToolID]) -> List[Any]:
-        tools: List[cls] = []
+    def _convert_to(target_type, tool_database: Union[List[ToolID], List[SystemID]]) -> List[Any]:
```
Comment (arnaudgelas) [inline backend/koala/api.py:96]: @verthais this method is giving some trouble with `mypy`

**Comment by arnaudgelas** [comment]
@verthais Consider running `isort`

File: backend/koala/api.py (hunk 2 above)
Comment (arnaudgelas) [reply backend/koala/api.py:169]: leave it for now

**Review by mymichu — APPROVED** [review]

**Pull request merged**